### PR TITLE
Adding support for new  endpoints which expose Elasticsearch's _mappi…

### PR DESCRIPTION
**Pull request by maxsibilla**
…ng endpoint

### CODE REVIEW

**Comment by maxsibilla** [comment]
Related Search API PR to modify YAML spec https://github.com/sennetconsortium/search-api/pull/197

File: src/app.py [hunk 1]
```diff
@@ -241,6 +250,19 @@ def search(self):
 
         return response
 
+    # Defaults to using the consortium index. Exposes Elasticsearch's `_mapping` endpoint to return details
+    # regarding a specific index.
+    def mapping(self):
+        return self._get_index_mappings('entities')
```
Comment (yuanzhou) [inline src/app.py:256]: Use the configured value from `self.INDICES['default_index']` instead of the hardcoded one.

File: src/app.py [hunk 2]
```diff
@@ -76,6 +77,14 @@ def __index():
         def __search():
             return self.search()
 
+        @self.app.route('/mapping', methods=['GET'])
+        def __mapping():
+            return self.mapping()
+
+        @self.app.route('/mapping/<index_name>', methods=['GET'])
```
Comment (yuanzhou) [inline src/app.py:84]: To be consistent with other endpoints, we should use `/<index_without_prefix>/mapping`.

File: src/app.py [hunk 3]
```diff
@@ -241,6 +250,19 @@ def search(self):
...
+    # Defaults to using the consortium index. Exposes Elasticsearch's `_mapping` endpoint to return details
+    # regarding a specific index.
+    def mapping(self):
+        return self._get_index_mappings('entities')
+
+    # Exposes Elasticsearch's `_mapping` endpoint to return details regarding a specific index.
+    def mapping_index(self, index_name):
+        # Check that the provided index exists in the configuration
+        if index_name not in self.INDICES['indices'].keys():
+            return f"Unable to find index '{index_name}'.", 400
```
Comment (yuanzhou) [inline src/app.py:262]: Should be a 404 status.

**Review by yuanzhou — CHANGES_REQUESTED** [review]
@maxsibilla you may need to update the Smart API yaml and AWS API Gateway definitions for the actual SenNet search-api. I'll take care of the one for HuBMAP.

**Comment by maxsibilla** [comment]
@yuanzhou Those changes have been made. Also I already have the modifications to the Smart API here https://github.com/sennetconsortium/search-api/pull/197/files

**Pull request merged**